presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroupManager.java

```diff
@@ -204,7 +204,7 @@ public void loadConfigurationManager()
                     MAX_QUEUED_QUERIES, Integer.toString(queryManagerConfig.getMaxQueuedQueries()));
             setConfigurationManager(LegacyResourceGroupConfigurationManager.NAME, legacyProperties);
         }
-        isConfigurationManagerLoaded = true;
+        isConfigurationManagerLoaded = new AtomicBoolean(true);
```

Comment:
```suggestion
        isConfigurationManagerLoaded.set(true);
```

If you reassign it, you'll have the same visibility problem.